MultimodalQnA/ui/gradio/multimodalqna_ui_gradio.py

```diff
@@ -315,6 +315,64 @@ def ingest_image_gen_caption(filepath, request: gr.Request):
     return
 
 
+def ingest_image_with_text(filepath, text, request: gr.Request):
+    yield (gr.Textbox(visible=True, value="Please wait for your uploaded image to be ingested into the database..."))
+    verified_filepath = os.path.normpath(filepath)
+    if not verified_filepath.startswith(tmp_upload_folder):
+        print("Found malicious image file name!")
+        yield (
+            gr.Textbox(
+                visible=True,
+                value="Your uploaded image's file name has special characters that are not allowed. Please consider updating the file name!",
```

Comment:
Can we say which special characters are disallowed? And maybe personal preference, but I would change the `!` to a `.`

Reply:
I was trying to change the original as little as possible, but I could make this improvement without too much trouble.